### PR TITLE
v1.7.0-beta.1

### PR DESCRIPTION
**Pull request by ahk**
## Launch Checklist

 - [x] briefly describe the changes in this PR
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,37 @@
...
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
+* Use `browser.devicePixelRatio` instead of directly calling `window.devicePixelRatio` ([#9063](https://github.com/mapbox/mapbox-gl-js/pull/9063))
+* Add instrumentation for performance metrics tracking ([#9035](https://github.com/mapbox/mapbox-gl-js/pull/9035))
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
+* Add `derefLayers` to the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
+
+## 🍏 Improvements
+* Update Code of Conduct (v2.0, https links) ([#9176](https://github.com/mapbox/mapbox-gl-js/pull/9176))
+* Optimize `findLoadedParent` for rendering view with a large number of raster tiles ([#9050](https://github.com/mapbox/mapbox-gl-js/pull/9050))
+* `ImageBitmap` and `OffscreenCanvas` for DEM tiles ([#8845](https://github.com/mapbox/mapbox-gl-js/pull/8845))
```
Comment (arindam1993) [inline CHANGELOG.md]: I think I would reword this as :
`Improve performance for hillshade and raster layers by implementing a progressive enhancement that utilizes `ImageBitmap` and `OffscreenCanvas`.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+
+## ✨ Features
+* Add a new constructor option to `mapboxgl.Popup`, `closeOnMove`, that closes the popup when the map's position changes ([#9163](https://github.com/mapbox/mapbox-gl-js/pull/9163))
+* Add third parameter to once event handler ([#8875](https://github.com/mapbox/mapbox-gl-js/pull/8875))
+* Allow programmatic construction of style even when one is not passed in initially ( h/t @stepankuzmin ) ([#8924](https://github.com/mapbox/mapbox-gl-js/pull/8924))
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
+* Use `browser.devicePixelRatio` instead of directly calling `window.devicePixelRatio` ([#9063](https://github.com/mapbox/mapbox-gl-js/pull/9063))
+* Add instrumentation for performance metrics tracking ([#9035](https://github.com/mapbox/mapbox-gl-js/pull/9035))
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
+* Add `derefLayers` to the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
```
Comment (arindam1993) [inline CHANGELOG.md]:
```
Export the internal `derefLayers` utility function from the `style-spec` bundle.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,37 @@
+## 1.7.0-beta.1
+
+## ✨ Features
+* Add a new constructor option to `mapboxgl.Popup`, `closeOnMove`, that closes the popup when the map's position changes ([#9163](https://github.com/mapbox/mapbox-gl-js/pull/9163))
+* Add third parameter to once event handler ([#8875](https://github.com/mapbox/mapbox-gl-js/pull/8875))
+* Allow programmatic construction of style even when one is not passed in initially ( h/t @stepankuzmin ) ([#8924](https://github.com/mapbox/mapbox-gl-js/pull/8924))
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
+* Use `browser.devicePixelRatio` instead of directly calling `window.devicePixelRatio` ([#9063](https://github.com/mapbox/mapbox-gl-js/pull/9063))
+* Add instrumentation for performance metrics tracking ([#9035](https://github.com/mapbox/mapbox-gl-js/pull/9035))
```
Comment (arindam1993) [inline CHANGELOG.md]: Don't think needs a changelog entry, its kinda internal.

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,37 @@
+## 1.7.0-beta.1
+
+## ✨ Features
+* Add a new constructor option to `mapboxgl.Popup`, `closeOnMove`, that closes the popup when the map's position changes ([#9163](https://github.com/mapbox/mapbox-gl-js/pull/9163))
+* Add third parameter to once event handler ([#8875](https://github.com/mapbox/mapbox-gl-js/pull/8875))
+* Allow programmatic construction of style even when one is not passed in initially ( h/t @stepankuzmin ) ([#8924](https://github.com/mapbox/mapbox-gl-js/pull/8924))
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
+* Use `browser.devicePixelRatio` instead of directly calling `window.devicePixelRatio` ([#9063](https://github.com/mapbox/mapbox-gl-js/pull/9063))
```
Comment (arindam1993) [inline CHANGELOG.md]: This isn't really a feature , right?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,37 @@
+## 1.7.0-beta.1
+
+## ✨ Features
+* Add a new constructor option to `mapboxgl.Popup`, `closeOnMove`, that closes the popup when the map's position changes ([#9163](https://github.com/mapbox/mapbox-gl-js/pull/9163))
+* Add third parameter to once event handler ([#8875](https://github.com/mapbox/mapbox-gl-js/pull/8875))
```
Comment (arindam1993) [inline CHANGELOG.md]:
```
`map.once()` now allows specifying a layer-idm as a third parameter making it consistent with `map.on()`.
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,37 @@
...
+* Use `browser.devicePixelRatio` instead of directly calling `window.devicePixelRatio` ([#9063](https://github.com/mapbox/mapbox-gl-js/pull/9063))
+* Add instrumentation for performance metrics tracking ([#9035](https://github.com/mapbox/mapbox-gl-js/pull/9035))
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
+* Add `derefLayers` to the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
+
+## 🍏 Improvements
+* Update Code of Conduct (v2.0, https links) ([#9176](https://github.com/mapbox/mapbox-gl-js/pull/9176))
+* Optimize `findLoadedParent` for rendering view with a large number of raster tiles ([#9050](https://github.com/mapbox/mapbox-gl-js/pull/9050))
+* `ImageBitmap` and `OffscreenCanvas` for DEM tiles ([#8845](https://github.com/mapbox/mapbox-gl-js/pull/8845))
+* Use stencil test instead of tile mask approach ([#9012](https://github.com/mapbox/mapbox-gl-js/pull/9012))
```
Comment (arindam1993) [inline CHANGELOG.md]:
```
Improve performance for raster tile rendering by using the stencil buffer.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,37 @@
...
+* Fix URL hash with no bearing causing map to fail to load ([#9170](https://github.com/mapbox/mapbox-gl-js/pull/9170))
+* Fix bug in `GeolocateControl` where multiple instances of the control on one page may result in the user location not being updated ([#9092](https://github.com/mapbox/mapbox-gl-js/pull/9092))
+* Fix query `fill-extrusions` made from polygons with coincident points and polygons with less than four points ([#9138](https://github.com/mapbox/mapbox-gl-js/pull/9138))
+* Fix bug where `symbol-sort-key` was not used for collisions that crossed tile boundaries ([#9054](https://github.com/mapbox/mapbox-gl-js/pull/9054))
+* Fix `versions` benchmark not being able to fetch latest release ([#9084](https://github.com/mapbox/mapbox-gl-js/pull/9084))
+* Support `ImageBitmap` in `Map#addImage` and `Map#updateImage` ([#9139](https://github.com/mapbox/mapbox-gl-js/pull/9139))
+* Fix bug in `DragRotateHandler._onMouseUp` getting stuck in drag/rotate ([#9137](https://github.com/mapbox/mapbox-gl-js/pull/9137))
+* Update quads.js algorithm comments ([#9128](https://github.com/mapbox/mapbox-gl-js/pull/9128)) (h/t [cs09g](https://github.com/cs09g))
```
Comment (arindam1993) [inline CHANGELOG.md]: There's npothing user-facing here, it was a variable name change.

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,37 @@
...
+* Fix validation error messages in the browser ([#9073](https://github.com/mapbox/mapbox-gl-js/pull/9073))
+* Prevent changing bearing via URL hash when rotation is disabled ([#9156](https://github.com/mapbox/mapbox-gl-js/pull/9156))
+* Fix URL hash with no bearing causing map to fail to load ([#9170](https://github.com/mapbox/mapbox-gl-js/pull/9170))
+* Fix bug in `GeolocateControl` where multiple instances of the control on one page may result in the user location not being updated ([#9092](https://github.com/mapbox/mapbox-gl-js/pull/9092))
+* Fix query `fill-extrusions` made from polygons with coincident points and polygons with less than four points ([#9138](https://github.com/mapbox/mapbox-gl-js/pull/9138))
+* Fix bug where `symbol-sort-key` was not used for collisions that crossed tile boundaries ([#9054](https://github.com/mapbox/mapbox-gl-js/pull/9054))
+* Fix `versions` benchmark not being able to fetch latest release ([#9084](https://github.com/mapbox/mapbox-gl-js/pull/9084))
+* Support `ImageBitmap` in `Map#addImage` and `Map#updateImage` ([#9139](https://github.com/mapbox/mapbox-gl-js/pull/9139))
```
Comment (arindam1993) [inline CHANGELOG.md]: This can be deleted.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,3 +1,37 @@
...
+
+## 🐞 Bug Fixes
+* Change the type of tile id key to string to prevent hash collisions ([#8979](https://github.com/mapbox/mapbox-gl-js/pull/8979))
+* Fix validation error messages in the browser ([#9073](https://github.com/mapbox/mapbox-gl-js/pull/9073))
+* Prevent changing bearing via URL hash when rotation is disabled ([#9156](https://github.com/mapbox/mapbox-gl-js/pull/9156))
+* Fix URL hash with no bearing causing map to fail to load ([#9170](https://github.com/mapbox/mapbox-gl-js/pull/9170))
+* Fix bug in `GeolocateControl` where multiple instances of the control on one page may result in the user location not being updated ([#9092](https://github.com/mapbox/mapbox-gl-js/pull/9092))
+* Fix query `fill-extrusions` made from polygons with coincident points and polygons with less than four points ([#9138](https://github.com/mapbox/mapbox-gl-js/pull/9138))
+* Fix bug where `symbol-sort-key` was not used for collisions that crossed tile boundaries ([#9054](https://github.com/mapbox/mapbox-gl-js/pull/9054))
+* Fix `versions` benchmark not being able to fetch latest release ([#9084](https://github.com/mapbox/mapbox-gl-js/pull/9084))
```
Comment (arindam1993) [inline CHANGELOG.md]: This is for tooling, I don't think it needs a changelog entry.

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,3 +1,37 @@
...
+* Update Code of Conduct (v2.0, https links) ([#9176](https://github.com/mapbox/mapbox-gl-js/pull/9176))
+* Optimize `findLoadedParent` for rendering view with a large number of raster tiles ([#9050](https://github.com/mapbox/mapbox-gl-js/pull/9050))
+* `ImageBitmap` and `OffscreenCanvas` for DEM tiles ([#8845](https://github.com/mapbox/mapbox-gl-js/pull/8845))
+* Use stencil test instead of tile mask approach ([#9012](https://github.com/mapbox/mapbox-gl-js/pull/9012))
+* Update `symbol-avoid-edges` documentation to acknowledge the existence of global collision detection ([#9157](https://github.com/mapbox/mapbox-gl-js/pull/9157))
+* Remove reference to deprecated `in` expression ([#9102](https://github.com/mapbox/mapbox-gl-js/pull/9102))
+
+## 🐞 Bug Fixes
+* Change the type of tile id key to string to prevent hash collisions ([#8979](https://github.com/mapbox/mapbox-gl-js/pull/8979))
+* Fix validation error messages in the browser ([#9073](https://github.com/mapbox/mapbox-gl-js/pull/9073))
```
Comment (arindam1993) [inline CHANGELOG.md]:
```
Fix validation error messages not being displayed when using the browser bundle.
```
Also, I think this belongs in the  this belongs in the style-spec changelog.

File: src/style-spec/CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,15 @@
+## 13.11.0-beta.1
+
+### ✨ Features and improvements
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
```
Comment (arindam1993) [inline src/style-spec/CHANGELOG.md]: Duplicated in the gl-js changelog

File: src/style-spec/CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,15 @@
...
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
+* Add `derefLayers` to the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
+* Update `symbol-avoid-edges` documentation regarding global collision detection ([#9157](https://github.com/mapbox/mapbox-gl-js/pull/9157))
+* Remove reference to deprecated `in` expression ([#9102](https://github.com/mapbox/mapbox-gl-js/pull/9102))
+
+### 🐛 Bug fixes
+* Fix bug where `symbol-sort-key` was not used for collisions that crossed tile boundaries ([#9054](https://github.com/mapbox/mapbox-gl-js/pull/9054))
```
Comment (arindam1993) [inline src/style-spec/CHANGELOG.md]: Actually a bunch of these are duplicated.

File: src/style-spec/CHANGELOG.md (hunk 11 above)
Comment (ahk) [reply src/style-spec/CHANGELOG.md]: I think they're intended to be duplicated, according to the process documentation.

File: CHANGELOG.md (hunk 4 above)
Comment (ahk) [reply CHANGELOG.md]: I think it is because it introduces the ability to:

1. show map inside scaled element (transform: scale) with working mouse events
2. change map rendering pixel ratio.

1 could be considered a bug but is featurey to me, I think 2 is definitely a feature.

File: CHANGELOG.md (hunk 5 above)
Comment (ahk) [reply CHANGELOG.md]: Added as 'layer id' instead of layer-idm

File: CHANGELOG.md (hunk 10 above)
Comment (ahk) [reply CHANGELOG.md]: Yep, good catch.

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,3 +1,33 @@
+## 1.7.0-beta.1
+
+## ✨ Features
+* Add a new constructor option to `mapboxgl.Popup`, `closeOnMove`, that closes the popup when the map's position changes ([#9163](https://github.com/mapbox/mapbox-gl-js/pull/9163))
+* `map.once()` now allows specifying a layer id as a third parameter making it consistent with `map.on()` ([#8875](https://github.com/mapbox/mapbox-gl-js/pull/8875))
+* Allow programmatic construction of style even when one is not passed in initially ( h/t @stepankuzmin ) ([#8924](https://github.com/mapbox/mapbox-gl-js/pull/8924))
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
```
Comment (mourner) [inline CHANGELOG.md]: I'd reorder the points so that bigger / more impactful ones are at the top. E.g. this seems to be the biggest feature of the release.

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,3 +1,33 @@
+## 1.7.0-beta.1
+
+## ✨ Features
+* Add a new constructor option to `mapboxgl.Popup`, `closeOnMove`, that closes the popup when the map's position changes ([#9163](https://github.com/mapbox/mapbox-gl-js/pull/9163))
+* `map.once()` now allows specifying a layer id as a third parameter making it consistent with `map.on()` ([#8875](https://github.com/mapbox/mapbox-gl-js/pull/8875))
+* Allow programmatic construction of style even when one is not passed in initially ( h/t @stepankuzmin ) ([#8924](https://github.com/mapbox/mapbox-gl-js/pull/8924))
```
Comment (mourner) [inline CHANGELOG.md]: This sounds a little cryptic, can we simplify the wording? E.g. "Allow creating a map without a style (an empty one will be created automatically)".

File: CHANGELOG.md (hunk 4 above)
Comment (mourner) [reply CHANGELOG.md]: You can add "... to be able to override it" to signify that it's a feature.

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,3 +1,33 @@
...
+* Export the internal `derefLayers` utility function from the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
+
+## 🍏 Improvements
+* Update Code of Conduct (v2.0, https links) ([#9176](https://github.com/mapbox/mapbox-gl-js/pull/9176))
+* Optimize `findLoadedParent` for rendering view with a large number of raster tiles ([#9050](https://github.com/mapbox/mapbox-gl-js/pull/9050))
+* Improve performance for hillshade and raster layers by implementing a progressive enhancement that utilizes `ImageBitmap` and `OffscreenCanvas` ([#8845](https://github.com/mapbox/mapbox-gl-js/pull/8845))
+* Improve performance for raster tile rendering by using the stencil buffer ([#9012](https://github.com/mapbox/mapbox-gl-js/pull/9012))
+* Update `symbol-avoid-edges` documentation to acknowledge the existence of global collision detection ([#9157](https://github.com/mapbox/mapbox-gl-js/pull/9157))
+* Remove reference to deprecated `in` expression ([#9102](https://github.com/mapbox/mapbox-gl-js/pull/9102))
```
Comment (mourner) [inline CHANGELOG.md]: This tripped me up — we just introduced the `in` expression and this point makes it sound like we deprecated it. 😅 Needs rewording, or maybe just remove from changelog.

Comment (ahk) [reply CHANGELOG.md]: Removed.

File: CHANGELOG.md (hunk 4 above)
Comment (ahk) [reply CHANGELOG.md]: Done.

File: CHANGELOG.md (hunk 13 above)
Comment (ahk) [reply CHANGELOG.md]: Ok, changed. Is this a better order?

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,3 +1,32 @@
...
+## 🍏 Improvements
+* Update Code of Conduct (v2.0, https links) ([#9176](https://github.com/mapbox/mapbox-gl-js/pull/9176))
+* Optimize `findLoadedParent` for rendering view with a large number of raster tiles ([#9050](https://github.com/mapbox/mapbox-gl-js/pull/9050))
+* Improve performance for hillshade and raster layers by implementing a progressive enhancement that utilizes `ImageBitmap` and `OffscreenCanvas` ([#8845](https://github.com/mapbox/mapbox-gl-js/pull/8845))
+* Improve performance for raster tile rendering by using the stencil buffer ([#9012](https://github.com/mapbox/mapbox-gl-js/pull/9012))
+* Update `symbol-avoid-edges` documentation to acknowledge the existence of global collision detection ([#9157](https://github.com/mapbox/mapbox-gl-js/pull/9157))
+
+## 🐞 Bug Fixes
+* Change the type of tile id key to string to prevent hash collisions ([#8979](https://github.com/mapbox/mapbox-gl-js/pull/8979))
+* Fix validation error messages in the browser ([#9073](https://github.com/mapbox/mapbox-gl-js/pull/9073))
```
Comment (arindam1993) [inline CHANGELOG.md]: Looked at the `1.6.1` changelog, this was already mentioned there, I think it can be deleted.

File: src/style-spec/CHANGELOG.md [hunk 17]
```diff
@@ -1,3 +1,16 @@
...
+### ✨ Features and improvements
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
+* Export the internal `derefLayers` utility function from the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
+* Update `symbol-avoid-edges` documentation regarding global collision detection ([#9157](https://github.com/mapbox/mapbox-gl-js/pull/9157))
+* Remove reference to deprecated `in` expression ([#9102](https://github.com/mapbox/mapbox-gl-js/pull/9102))
+
+### 🐛 Bug fixes
+* Fix bug where `symbol-sort-key` was not used for collisions that crossed tile boundaries ([#9054](https://github.com/mapbox/mapbox-gl-js/pull/9054))
+* Fix validation error messages in the browser ([#9073](https://github.com/mapbox/mapbox-gl-js/pull/9073))
```
Comment (arindam1993) [inline src/style-spec/CHANGELOG.md]: This is in the `13.10.2` style spec changelog already,

File: src/style-spec/CHANGELOG.md [hunk 18]
```diff
@@ -1,3 +1,16 @@
...
+
+### ✨ Features and improvements
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
+* Export the internal `derefLayers` utility function from the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
+* Update `symbol-avoid-edges` documentation regarding global collision detection ([#9157](https://github.com/mapbox/mapbox-gl-js/pull/9157))
+* Remove reference to deprecated `in` expression ([#9102](https://github.com/mapbox/mapbox-gl-js/pull/9102))
+
+### 🐛 Bug fixes
+* Fix bug where `symbol-sort-key` was not used for collisions that crossed tile boundaries ([#9054](https://github.com/mapbox/mapbox-gl-js/pull/9054))
```
Comment (arindam1993) [inline src/style-spec/CHANGELOG.md]: This isn't really a style-spec change, right?

File: src/style-spec/CHANGELOG.md [hunk 19]
```diff
@@ -1,3 +1,16 @@
+## 13.11.0-beta.1
+
+### ✨ Features and improvements
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
+* Export the internal `derefLayers` utility function from the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
+* Update `symbol-avoid-edges` documentation regarding global collision detection ([#9157](https://github.com/mapbox/mapbox-gl-js/pull/9157))
+* Remove reference to deprecated `in` expression ([#9102](https://github.com/mapbox/mapbox-gl-js/pull/9102))
```
Comment (arindam1993) [inline src/style-spec/CHANGELOG.md]: Same thing as before as @mourner , mentioned about it being confusing that we just introduced it, so this is weird.

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,3 +1,32 @@
+## 1.7.0-beta.1
+
+## ✨ Features
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
+* Use `browser.devicePixelRatio` instead of directly calling `window.devicePixelRatio` to be able to override it ([#9063](https://github.com/mapbox/mapbox-gl-js/pull/9063))
```
Comment (asheemmamoowala) [inline CHANGELOG.md]: What is the developer facing change here? Is it that developers can hijack `window.devicePixelRatio` and GL-JS will correctly get that update? cc @kkaefer 

Also missing h/t to pakastin

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,3 +1,32 @@
...
+* Add a new constructor option to `mapboxgl.Popup`, `closeOnMove`, that closes the popup when the map's position changes ([#9163](https://github.com/mapbox/mapbox-gl-js/pull/9163))
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
+* Allow creating a map without a style (an empty one will be created automatically) ( h/t @stepankuzmin ) ([#8924](https://github.com/mapbox/mapbox-gl-js/pull/8924))
+* `map.once()` now allows specifying a layer id as a third parameter making it consistent with `map.on()` ([#8875](https://github.com/mapbox/mapbox-gl-js/pull/8875))
+* Export the internal `derefLayers` utility function from the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
+
+## 🍏 Improvements
+* Update Code of Conduct (v2.0, https links) ([#9176](https://github.com/mapbox/mapbox-gl-js/pull/9176))
```
Comment (asheemmamoowala) [inline CHANGELOG.md]: Does this need to be in the release? It impacts development and contributors but not the release specifically.

File: CHANGELOG.md [hunk 22]
```diff
@@ -1,3 +1,32 @@
...
+* Optimize `findLoadedParent` for rendering view with a large number of raster tiles ([#9050](https://github.com/mapbox/mapbox-gl-js/pull/9050))
```
Comment (asheemmamoowala) [inline CHANGELOG.md]: The customer facing improvement 
```suggestion
* Improve performance of raster layers on large screens ([#9050](https://github.com/mapbox/mapbox-gl-js/pull/9050))
```

File: src/style-spec/CHANGELOG.md [hunk 23]
```diff
@@ -1,3 +1,16 @@
+## 13.11.0-beta.1
+
+### ✨ Features and improvements
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
```
Comment (asheemmamoowala) [inline src/style-spec/CHANGELOG.md]:
```suggestion
* Add `promoteId` option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
```

File: CHANGELOG.md [hunk 24]
```diff
@@ -1,3 +1,32 @@
+## 1.7.0-beta.1
+
+## ✨ Features
+* Add option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
```
Comment (asheemmamoowala) [inline CHANGELOG.md]:
```suggestion
* Add `promoteId` option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
```

File: src/style-spec/CHANGELOG.md (hunk 19 above)
Comment (asheemmamoowala) [inline src/style-spec/CHANGELOG.md]: How about this instead: 
```suggestion
* Remove reference to `in` function which has been replaced by the `in` expression ([#9102](https://github.com/mapbox/mapbox-gl-js/pull/9102))
```

File: CHANGELOG.md (hunk 16 above)
Comment (ahk) [reply CHANGELOG.md]: Maybe this is an issue with the cherry-pick detection logic? It's kinda brittle because it relies on the commit message being exactly the same.

Logic should be If something is on master since the last release branch it's included, but things found to have been cherry picked into the last release should be filtered out.

File: CHANGELOG.md (hunk 21 above)
Comment (ahk) [reply CHANGELOG.md]: My thinking here was that it was a notable change worth advertising, but agree it doesn't affect the release from a technical perspective. Can remove.

File: CHANGELOG.md (hunk 16 above)
Comment (ahk) [reply CHANGELOG.md]: Gonna investigate this further.

File: src/style-spec/CHANGELOG.md (hunk 18 above)
Comment (ahk) [reply src/style-spec/CHANGELOG.md]: My thinking was that people who care about the style would care about this because it changes the behavior of `symbol-sort-key`. It's a bug in that the style-spec documentation misspecified the behavior, but now that the behavior matches the documentation it's a mildly breaking change to the semantics of the layout property.

File: CHANGELOG.md (hunk 20 above)
Comment (ahk) [reply CHANGELOG.md]: Fixed the h/t. I had misunderstood the user facing change here a bit, not exactly sure how to summarize. My best understanding is that putting this on `browser` means you can override it, since it's not a read-only property on the window. I think this allows you to hack around an independently scaled map element to fix other downstream issues, which we are also working on supporting as separate features and fixes. Any advice here?

File: src/style-spec/CHANGELOG.md (hunk 19 above)
Comment (ahk) [reply src/style-spec/CHANGELOG.md]: I'm going to accept @asheemmamoowala suggestion and also duplicate that into the main changelog.

File: CHANGELOG.md [hunk 25]
```diff
@@ -1,3 +1,31 @@
...
+* Use `browser.devicePixelRatio` instead of directly calling `window.devicePixelRatio` to be able to override it ([#9063](https://github.com/mapbox/mapbox-gl-js/pull/9063)) (h/t [pakastin](https://github.com/pakastin)) 
+* Add a new constructor option to `mapboxgl.Popup`, `closeOnMove`, that closes the popup when the map's position changes ([#9163](https://github.com/mapbox/mapbox-gl-js/pull/9163))
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
+* Allow creating a map without a style (an empty one will be created automatically) ( h/t @stepankuzmin ) ([#8924](https://github.com/mapbox/mapbox-gl-js/pull/8924))
+* `map.once()` now allows specifying a layer id as a third parameter making it consistent with `map.on()` ([#8875](https://github.com/mapbox/mapbox-gl-js/pull/8875))
+* Export the internal `derefLayers` utility function from the `style-spec` bundle ([#9121](https://github.com/mapbox/mapbox-gl-js/pull/9121)) (h/t [ahocevar](https://github.com/ahocevar))
```
Comment (arindam1993) [inline CHANGELOG.md]: I dunno but this still feels like it shouldn't belong in gl-js features.

File: CHANGELOG.md [hunk 26]
```diff
@@ -1,3 +1,31 @@
+## 1.7.0-beta.1
+
+## ✨ Features
+* Add `promoteId` option to use a feature property as ID for feature state ([#8987](https://github.com/mapbox/mapbox-gl-js/pull/8987))
+* Use `browser.devicePixelRatio` instead of directly calling `window.devicePixelRatio` to be able to override it ([#9063](https://github.com/mapbox/mapbox-gl-js/pull/9063)) (h/t [pakastin](https://github.com/pakastin)) 
+* Add a new constructor option to `mapboxgl.Popup`, `closeOnMove`, that closes the popup when the map's position changes ([#9163](https://github.com/mapbox/mapbox-gl-js/pull/9163))
+* Edit convert functions to output linear interpolation ([#9107](https://github.com/mapbox/mapbox-gl-js/pull/9107)) (h/t [Jo-IE](https://github.com/Jo-IE))
```
Comment (arindam1993) [inline CHANGELOG.md]: This probably needs a bit more context, or it can be deleted, from reading the original issue I don't think it changes anything.

Comment (ahk) [reply CHANGELOG.md]: Yeah, sorry I had a lot of issues to scan through and definitely didn't get all the details right. This was one of the ones I felt the most hazy about. No apologies needed thank you so much for this help! Can delete, have been leaning on the side of leaving in things with hattips.

Comment (ahk) [reply CHANGELOG.md]: Also removing from style-spec changelog.

File: CHANGELOG.md (hunk 25 above)
Comment (ahk) [reply CHANGELOG.md]: Removed.

File: CHANGELOG.md (hunk 20 above)
Comment (ahk) [reply CHANGELOG.md]: Ok, I gave this another try to say more about why someone would use this. Good for beta?

Comment (ahk) [reply CHANGELOG.md]: Confused about how to word this and the intention of the patch, sorry. Going to remove this for now and we can add back in the final release if needed.
